### PR TITLE
Fix #87 - Push images to DockerHub and don't use GCR

### LINKED ISSUE #87: Publish a branch to DockerHub
Use the environment variables DOCKERHUB_REPO, DOCKER_USER, and DOCKER_PASS in your CircleCI job config. You can then reference those to push to Dockerhub. See https://github.com/mozilla-services/Dockerflow/blob/master/.circleci/config.yml#L68-L82 for an example.

### PR DESCRIPTION
**Pull request by jcjones**
This produces images at Docker.io for 
- tagged builds, see: https://hub.docker.com/r/jcjones/crlite/tags?name=v0.0.0
- untagged builds on the staging branch, see: https://hub.docker.com/r/jcjones/crlite/tags?name=98e104f47d84d2793a4b6161153fcd3f813b0f4d

It splits out two workflows and expects the following environment variables be set:
- DOCKER_IMAGE: jcjones/crlite
- DOCKER_LOGIN
- DOCKER_PASSWORD

Remaining todos:
- [x] Pushing to a branch is triggering the tagged builds workflow, so exclude that

### CODE REVIEW

**Comment by jcjones** [comment]
Works well.

**Comment by smarnach** [comment]
This looks good to me.

It looks like the [docker orb uses DOCKER_LOGIN and DOCKER_PASSWORD for the credentials by default](https://circleci.com/orbs/registry/orb/circleci/docker#commands-check). I can set these in CircleCI if you give me permission to do so. I believe I need write access to this repo to be allowed to make these changes.

The new repository location on Dockerhub will be mozilla/crlite.

File: .circleci/config.yml [hunk 1]
```diff
@@ -2,10 +2,76 @@ version: 2.1
...
+          the environment variable to use as a tag
+        type: string
+
+    steps:
+      - docker/check
+      - build-and-publish:
+          path: containers
+          tag: <<parameters.tag-param>>
+      - docker/update-description:
+          image: ${DOCKER_IMAGE}
```
Comment (jcjones) [inline .circleci/config.yml:53]: @smarnach Experimentation indicates that these lines require DOCKER_PASSWORD to be an actual user password, not a service key. Which will be used for the ops configuration?

**Comment by jcjones** [comment]
> This looks good to me.
> 
> It looks like the [docker orb uses DOCKER_LOGIN and DOCKER_PASSWORD for the credentials by default](https://circleci.com/orbs/registry/orb/circleci/docker#commands-check). I can set these in CircleCI if you give me permission to do so. I believe I need write access to this repo to be allowed to make these changes.
> 
> The new repository location on Dockerhub will be mozilla/crlite.

You have write access to this repo. Check and see?

**Comment by smarnach** [comment]
All working well now! Looks like updating the description requires giving the user "Admin" permissions on the repo – "Read/Write" was enough for pushing the image, but not for updating the description. According to our documentation, we always use the password and not some API key. We create a dedicated user account which is only used by CI.

**Comment by smarnach** [comment]
I reverted the temporary commit disabling the filter on the push job. Here's the log of the successful push: https://circleci.com/gh/mozilla/crlite/402

**Comment by jcjones** [comment]
Great, thanks!

**Pull request merged**